libs/datasets/timeseries.py

```diff
@@ -26,6 +31,50 @@ def __str__(self):
         return f"DuplicateDataException({self.message})"
 
 
+@dataclass(frozen=True)
+class OneRegionTimeseriesDataset:
+    """A set of timeseries with values from one region."""
+
+    # Do not make an assumptions about a FIPS or locationID column in the DataFrame.
+    data: pd.DataFrame
+    # The region is not an attribute at this time because it simplifies making instances and
+    # code that needs the region of an instance already has it.
+
+    def __post_init__(self):
+        region_count = self.data[CommonFields.FIPS].nunique()
+        if region_count == 0:
+            _log.warning(f"Creating {self.__class__.__name__} with zero regions")
+        elif region_count != 1:
+            raise ValueError(
+                f"Does not have exactly one region: "
+                f"{self.data[CommonFields.FIPS].value_counts()}"
+            )
+
+    @property
+    def empty(self):
+        return self.data.empty
+
+    def has_one_region(self) -> bool:
+        return True
+
+    def yield_records(self) -> Iterable[dict]:
+        # Copied from dataset_base.py
+        # It'd be faster to use self.data.itertuples or find a way to avoid yield_records, but that
+        # needs larger changes in code calling this.
+        for idx, row in self.data.iterrows():
+            yield row.where(pd.notnull(row), None).to_dict()
+
+    def get_subset(self, after=None, columns=tuple()):
+        rows_key = dataset_utils.make_rows_key(self.data, after=after,)
+        columns_key = list(columns) if columns else slice(None, None, None)
+        return OneRegionTimeseriesDataset(
+            self.data.loc[rows_key, columns_key].reset_index(drop=True)
+        )
+
+    def remove_padded_nans(self, columns: List[str]):
+        return OneRegionTimeseriesDataset(_remove_padded_nans(self.data, columns))
```

Comment:
How about `self.__class__` rather than calling the static class name (in case of subclassing)?

Reply:
I tried but than we lose static type checking of the parameters used to make the class. I'll try marking it @final https://sobolevn.me/2018/07/real-python-contants